backend/src/common/plugins/authorization/authorization-plugin.ts

```diff
@@ -0,0 +1,46 @@
+import fp from 'fastify-plugin';
+import { type AuthApiPath, type HttpError } from 'shared/build/index.js';
+
+import { type AuthService } from '~/bundles/auth/auth.service';
+import { getToken } from '~/bundles/auth/helpers/helpers.js';
+import { ControllerHook } from '~/common/controller/enums/enums.js';
+import { type IService } from '~/common/interfaces/service.interface';
+
+type AuthorizationPluginPayload = {
+    publicRoutes: Partial<Record<AuthApiPath, string>>;
+    userService: IService;
+    authService: AuthService;
+};
+
+const authorization = fp<AuthorizationPluginPayload>(
+    async (
+        fastify,
+        { publicRoutes, authService, userService },
+    ): Promise<void> => {
+        fastify.decorateRequest('user', null);
+
+        fastify.addHook(ControllerHook.ON_REQUEST, async (request, reply) => {
+            try {
+                const isPublicRoute = Object.values(publicRoutes).includes(
+                    request.routerPath,
+                );
+
+                if (isPublicRoute) {
+                    return;
+                }
+                const token = getToken(request.headers);
+
+                const { id } =
+                    authService.verifyToken<Record<'id', string>>(token);
+
+                request.user = await userService.getUserWithProfile(id);
```

Comment:
It would be nice to have a more descriptive prop name, like authUser or currentUser